src/types/web3Permissions.ts

```diff
@@ -0,0 +1,30 @@
+import { TAddress } from '@types';
+
```

Comment:
This is a vendor type.
Please add a comment explaining where it's from (link to src if possible)
and what it's used for.